src/main/java/com/yelp/nrtsearch/server/utils/ArchiverImpl.java

```diff
@@ -69,20 +76,29 @@
   private final Tar tar;
   private final VersionManager versionManger;
   private final TransferManager transferManager;
+  private final ThreadPoolExecutor executor =
+      (ThreadPoolExecutor) Executors.newFixedThreadPool(NUM_S3_THREADS);
+  private final boolean downloadAsStream;
 
   @Inject
   public ArchiverImpl(
-      final AmazonS3 s3, final String bucketName, final Path archiverDirectory, final Tar tar) {
+      final AmazonS3 s3,
+      final String bucketName,
+      final Path archiverDirectory,
+      final Tar tar,
+      final boolean downloadAsStream) {
```

Comment:
nit: you could make a new constructor and use chaining. 

`this(s3, bucketName, archiverDirectory, tar, false)`

That way the test code could still call the existing constructor and not change code in tests.

Reply:
done